torch/nn/attention/_flex_attention.py

```diff
@@ -45,43 +52,74 @@ def _identity(
     return score
 
 
+def _no_mask(
+    batch: torch.Tensor,
+    head: torch.Tensor,
+    token_q: torch.Tensor,
+    token_kv: torch.Tensor,
+) -> torch.Tensor:
+    return token_q.new_ones(size=(), dtype=torch.bool, device=batch.device)
+
+
 _DEFAULT_SPARSE_BLOCK_SIZE = 128
 
 
 class BlockMask:
-    kv_num_blocks: torch.Tensor
-    kv_indices: torch.Tensor
-    q_num_blocks: torch.Tensor
-    q_indices: torch.Tensor
+    full_kv_num_blocks: torch.Tensor
```

Comment:
Nit: I think we should really add some documentation on this data structure should be interpreted. I know there is some by the kernel definition but I think it should be colocated here, or at least create a Note and reference it in relevant places

Reply:
Yes, I think we need do a pass to fill all necessary docs and comments.